File: java/src/main/java/com/penapereira/patterns/templatemethod/ConcreteClassA.java
package com.penapereira.patterns.templatemethod;

/** Implements the primitive operations and keeps the default hook. */
public final class ConcreteClassA extends AbstractClass {

    @Override
    protected String primitiveOperation1() {
        return "ConcreteClassA.primitiveOperation1";
    }

    @Override
    protected String primitiveOperation2() {
        return "ConcreteClassA.primitiveOperation2";
    }
}

File: typescript/src/template-method/concrete-class-a.ts
import { AbstractClass } from "./abstract-class.ts";

export class ConcreteClassA extends AbstractClass {
  protected primitiveOperation1(): string {
    return "ConcreteClassA.primitiveOperation1";
  }

  protected primitiveOperation2(): string {
    return "ConcreteClassA.primitiveOperation2";
  }
}
